### PR TITLE
feat(w3c/group): add `group` option to fetch WG details

### PR DESCRIPTION
**Pull request by sidvishnoi**
Closes https://github.com/w3c/respec/issues/2745
Wiki: https://github.com/w3c/respec/wiki/group

### CODE REVIEW

**Comment by sidvishnoi** [comment]
I realized we might want to support multiple groups. That will require a little change in respec.org API, or we can send multiple requests (which maybe fine given rarity of multiple groups)

File: src/w3c/group.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+const W3C_GROUPS_API = "https://respec.org/w3c/groups/";
+
+export async function run(conf) {
+  if (!conf.group) return;
+
+  const supersededOptions = ["wg", "wgURI", "wgId", "wgPatentURI"];
+  const usedSupersededOptions = supersededOptions.filter(opt => conf[opt]);
+  if (usedSupersededOptions.length) {
+    const outdatedOptionsStr = joinAnd(usedSupersededOptions, s => `\`${s}\``);
+    const msg = `Configuration options ${outdatedOptionsStr} are superseded by \`group\`. Please remove them from \`respecConfig\`.`;
```
Comment (marcoscaceres) [inline src/w3c/group.js:23]:
```suggestion
    const msg = `Configuration options ${outdatedOptionsStr} are superseded by \`group\` and will be overridden. Please remove them from \`respecConfig\`.`;
```

File: src/w3c/group.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+const W3C_GROUPS_API = "https://respec.org/w3c/groups/";
+
+export async function run(conf) {
+  if (!conf.group) return;
+
+  const supersededOptions = ["wg", "wgURI", "wgId", "wgPatentURI"];
+  const usedSupersededOptions = supersededOptions.filter(opt => conf[opt]);
+  if (usedSupersededOptions.length) {
+    const outdatedOptionsStr = joinAnd(usedSupersededOptions, s => `\`${s}\``);
+    const msg = `Configuration options ${outdatedOptionsStr} are superseded by \`group\` and will be overridden by ReSpec. Please remove them from \`respecConfig\`.`;
```
Comment (sidvishnoi) [inline src/w3c/group.js:23]: TODO: split this into two lines

File: src/w3c/group.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+  if (res.ok) {
+    const json = await res.json();
+    const { id: wgId, name: wg, URI: wgURI, patentURI: wgPatentURI } = json;
+    return { wg, wgId, wgURI, wgPatentURI };
+  }
+
+  if (res.status === 404) {
+    const text = await res.text();
+    pub("error", `Invalid value for \`respecConfig.group\`. ${text}`);
```
Comment (sidvishnoi) [inline src/w3c/group.js:64]:
```suggestion
    const msg = `No group with name \`"${group}"\` found.`;
    const hint = "Please add a [valid group name](https://github.com/w3c/respec/wiki/group) to `respecConfig.group`.";
    pub("error", `${msg} ${hint}`);
```

File: src/w3c/group.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  } else {
+    pub("error", `Failed to fetch group details (HTTP: ${res.status})`);
+  }
```
Comment (marcoscaceres) [inline src/w3c/group.js:67]: Maybe: 

```suggestion
  let msg = `Failed to fetch group details (HTTP: ${res.status})`;
  if (res.status === 404) {
    const text = await res.text();
    msg = `Couldn't find W3C group '\`respecConfig.group\`'. ${text}`;
  }
  pub("error", msg)
```

File: tests/spec/w3c/group-spec.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  it("overrides superseded options", async () => {
+    const inputConf = { group: "payments", wg: "foo", wgId: "1234" };
+    const conf = await getGroupConf(inputConf);
+    expect(conf.wg).not.toBe(inputConf.wg);
+    expect(conf.wgId).not.toBe(inputConf.wgId);
+    expect(conf.wgPatentURI).not.toBe(inputConf.wgPatentURI);
+    expect(conf.wgURI).not.toBe(inputConf.wgURI);
+  });
+
+  it("fails if invalid group name is specified", async () => {
```
Comment (marcoscaceres) [inline tests/spec/w3c/group-spec.js:51]:
```suggestion
  it("fails if invalid group name is not known, even if it looks right", async () => {
```

**Pull request merged**